.gitlab/ci/on-push.yml

```diff
@@ -263,3 +264,70 @@ slack-notify-nightly-build:
     # when triggering pipeline with alternate env variable value passed in the API call
     SLACK_CHANNEL: $SLACK_CHANNEL
     SLACK_JOB: 'true'
+
+
+.test_content_on_xsiam_server_instances_base:
+  tags:
+    - gke
+    - us-west1
+  extends:
+    - .default-job-settings
+    - .push-rule
+  variables:
+    ARTIFACTS_FOLDER: "${ARTIFACTS_FOLDER_MPV2}"
+    GCS_MARKET_BUCKET: "${GCS_MARKET_V2_BUCKET}"
+    ENV_RESULTS_PATH: "${ARTIFACTS_FOLDER_MPV2}/env_results.json"
+    SERVER_TYPE: "XSIAM"
+  needs: ["prepare-testing-bucket-mpv2"]
+  stage: run-instances
+  script:
+    - EXIT_CODE=0
+    - !reference [.download-demisto-conf]
+
+    - section_start "Lock XSIAM Machine"
+    - cp "$ARTIFACTS_FOLDER/filter_file.txt" "./artifacts/filter_file.txt"
+    - echo "Authenticating GCP"
+    - gcloud auth activate-service-account --key-file="$GCS_ARTIFACTS_KEY" > auth.out 2>&1
+    - echo "Auth done successfully"
+    - ./Tests/scripts/wait_in_line_for_xsiam_env.sh
+    - source XSIAMEnvVariables
+    - echo "XSIAM chosen_machine_id is $XSIAM_CHOSEN_MACHINE_ID"
+    - section_end "Lock XSIAM Machine"
+
+    - section_start "Clean XSIAM Machine"
+    - ./Tests/scripts/uninstall_packs_and_reset_bucket.sh
+    - section_end "Clean XSIAM Machine"
+
+    - section_start "Install Packs and run Test-Module"
+    - ./Tests/scripts/install_content_and_test_integrations.sh "$INSTANCE_ROLE" "$SERVER_TYPE"|| EXIT_CODE=$?
+    - section_end "Install Packs and run Test-Module"
+
+    - section_start "Run Tests"
+    - cp -f $ARTIFACTS_FOLDER/conf.json Tests/conf.json
+    - ./Tests/scripts/run_tests.sh "$INSTANCE_ROLE" || EXIT_CODE=$?
+    - |
+      if [[ -f ./Tests/failed_tests.txt ]]; then
+        cp ./Tests/failed_tests.txt $ARTIFACTS_FOLDER/failed_tests.txt
+      fi
+    - |
+      if [[ -f ./Tests/succeeded_tests.txt ]]; then
+        cp ./Tests/succeeded_tests.txt $ARTIFACTS_FOLDER/succeeded_tests.txt
+      fi
+    - section_end "Run Tests"
+    - exit $EXIT_CODE
+  after_script:
+    - echo "Job finished, removing lock file"
```

Comment:
if the test job fails this will still run? also what happens if this fails?

Reply:
If job fails this will run. If job cancelled, it wont run. 
If the lock files not deleted, the lock got free, since the script checks the status of build that owns lock file. 

To solve the issue that there can be too much unnecessary and not valid lock files, opened an [issue](https://github.com/demisto/etc/issues/47970).